packages/svelte/src/internal/client/runtime.js

```diff
@@ -191,10 +192,16 @@ export function check_dirtiness(reaction) {
 						// if linked to the dependency source – otherwise future updates will not be caught.
 						(dependency.reactions ??= []).push(reaction);
 					}
+				} else if (is_disconnected) {
+					reconnect_derived(reaction);
 				}
 			}
 		}
 
+		if (is_disconnected) {
+			reaction.f ^= DISCONNECTED;
+		}
+
```

Comment:
we can do this inside `reconnect_derived`

```suggestion
```

Reply:
Then we're doing this on each iteration of the loop?

Reply:
we are... but if that's incorrect then shouldn't `reconnect_derived` also be outside the loop?

Reply:
Ah gotcha.